### ISSUE TITLE
Iteraciones por segundo esperadas y cantidad de epocas

### ISSUE THREAD

**Issue by sara-gf** [JavierBZ/Tarea3-IIC3697#23]
Holaa,

Habiendo construido el entrenamiento, me preocupe de solo realizar operaciones de tensores sin iterar para acelerar todo. Ahora al momento de entrenar, propuse 20 epochs para tener 5 millones de pasos. Llenando el replay memory alcanza 900 it/s, y cuando pasa a entrenar esperaba que disminuyera, pero disminuyó a 19 it/s. Con eso me calcula un estimado de 3 hrs para terminar la epoch actual, lo que me hace estimar 60 hrs de entrenamiento, lo cual es demasiado. Ahora lo que pregunto no es como disminuir el tiempo, eso lo puedo investigar yo, lo que quiero saber es: a que velocidad de it/s deberíamos alcanzar?

**Comment by JavierBZ** [comment]
Hola, ya varios me han preguntado sobre como mejorar el rendimiento para llegar a un rendimiento óptimo mayor a 100 it/s, les puedo dar las siguientes sugerencias:

1. Optimicen la función sample del replay memory. Para ello les recomiendo usar la función random.sample, la cual permite obtener el batch de experiencias en a lo más dos lineas.
2. Para transformar a tensores utilicen torch.fom_numpy() en vez de torch.tensor(), es más eficiente.
3. Revisen bien y optimicen todas las operaciones de tensores que hagan tanto en la red Dueling DQN como en el cálculo de la loss
4. Por último, no esta de más recordarles que **no utilicen Fors!!!!**, cualquier uso de for les puede enlentecer mucho el rendimiento.

Espero les sirva

**Comment by sara-gf** [comment]
Después de estar hrs y hrs optimizando la función sample del replay de varias maneras sin ver aumentos en la velocidad, cambie el como transformar los tensores ocupando torch.from_numpy() y mi veolcidad aumento de 20 it/s a 180 it/s. Así que póngale ojo a los que no lo han hecho todavía.
